Update CI

zmoon opened the pull request:
* moving to the new micromamba Action
* [x] add more debug flags for GFortran
* [x] similar for Intel
* [x] DEBUG=2 option and document (29863dc9f)

zmoon committed:
Update CI

zmoon committed:
Update Python in lint workflow

zmoon committed:
More debug flags for GFortran

zmoon commented:
@drnimbusrain how do we want to handle FPEs? First one is here (case that `hcmref` is 0):
```diff
--- a/src/canopy_calcs.F90
+++ b/src/canopy_calcs.F90
@@ -78,8 +78,12 @@ SUBROUTINE canopy_calcs
 ! ... calculate wind speed from u and v
                 ubzref   = sqrt((uref**2.0) + (vref**2.0))

-! ... get scaled canopy model profile and sub-canopy layers
-                zhc         = zk/hcmref
+                ! ... get scaled canopy model profile and sub-canopy layers
+                if (hcmref > 0) then
+                    zhc = zk / hcmref
+                else
+                    zhc = 0
+                end if
                 cansublays  = floor(hcmref/modres)

 ! ... check for valid model vegetation types
```

drnimbusrain commented:
I agree with your first FPE change here.  Do you want to make this (and other) change to your branch as the FPEs come up with your improved debugging options?

drnimbusrain commented:
@zmoon Do you think you will be able to work on this again soon?  Thank you!

zmoon commented:
@drnimbusrain I'll try to work on it again soon.

drnimbusrain commented:
Thanks Zach...I added the additional debug flags and checked/solved the
FPEs in latest develop.

On Wed, Jan 24, 2024, 1:36 PM Zachary Moon ***@***.***> wrote:

> @drnimbusrain <https://github.com/drnimbusrain> I'll try to work on it
> again soon.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/noaa-oar-arl/canopy-app/pull/85#issuecomment-1908707187>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGLFYNQFPNHD3JUWKZGQXHLYQFIBDAVCNFSM6AAAAAA3HJFUT2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMBYG4YDOMJYG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

zmoon commented:
That's good, I'll merge that into here before working on it more then.

zmoon committed:
Merge branch 'develop' into update-ci

Initial DEBUG=2 option

some tweaks to flags, including setting error limit 0
for DEBUG=2

zmoon committed:
fmt

zmoon committed:
Gamma leaf age calc only if on

with the enhanced debug flags, was complaining about the
past (and/or current) LAI being unitialized in the first
expression of the calculation

zmoon committed:
DEBUG=2 in readme

zmoon committed:
Update actions/checkout

zmoon commented:
@drnimbusrain some notes:
* CI debug build still has DEBUG=1 not 2
* we're currently not testing all code paths, just the default namelist, so there may be other FPEs
* "array temporary was created" warnings we could alleviate by explicitly creating the temporary arrays. This is mainly for going from an array of floats to assigning to a float member of an array of derived type in the ncf read/write.

drnimbusrain commented:
@zmoon Thank you for working on this, and agree that more comprehensive checks of the different code pathways/options are needed to check for all FPEs.  I will try and work on this with your new debug options more thoroughly as we go along with more robust testing.  I think that @btang1 @MaggieMarvin @angehung5 could help with this DEBUG=2 testing of the different options for FPEs as well, since they are going to be focusing on different parts of code.   For now I approve, and we can improve later.

drnimbusrain commented:
I can also work on the "array temporary was created" issues too.  Thanks!

drnimbusrain commented on src/canopy_utils_mod.F90:
@zmoon Although, what has changed in the leafage option with your PR?

zmoon replied:
Should be nothing. I just moved the calculation under the if statement to avoid FPE related to the LAI being unitialized. Little note in d241beac46e

drnimbusrain reviewed (approved):
Looks OK, but just want to know about the leafage option changes here.

drnimbusrain commented:
OK, I will merge then and work with the DEBUG=2 option some to fix some other potential FPEs and the warnings...
